src/main/java/de/themoep/minedown/MineDownParser.java

```diff
@@ -504,16 +504,16 @@ public ComponentBuilder parseEvent(String text, String definitions) {
                 rainbowPhase = parsedRainbowPhase;
                 continue;
             } else if (!definition.contains("=")) {
-                Map<ChatColor, Boolean> parsed = parseColor(definition, "", true, backwardsCompatibility());
+                List<Map.Entry<ChatColor, Boolean>> parsed = parseColor(definition, "", true, backwardsCompatibility());
                 if (!parsed.isEmpty()) {
-                    for (Map.Entry<ChatColor, Boolean> e : parsed.entrySet()) {
+                    for (Map.Entry<ChatColor, Boolean> e : parsed) {
                         if (Util.isFormat(e.getKey())) {
                             formats.put(e.getKey(), e.getValue());
                         } else {
                             if (colors == null) {
-                                colors = new LinkedHashMap<>();
+                                colors = new ArrayList<>();
                             }
-                            colors.put(e.getKey(), e.getValue());
+                            colors.add(new AbstractMap.SimpleEntry<>(e.getKey(), e.getValue()));
```

Comment:
Pretty sure you can just add the entry directly into the colors list instead of creating a new entry as the entry values will never be modified.

Reply:
okay, shall I solve this?

Reply:
Yes, please.